src/main/kotlin/dev/pfilaretov42/aoc/aoc2023/day4/Scratchcards.kt

```diff
@@ -5,8 +5,37 @@ import kotlin.math.pow
 
 fun main() {
     val inputFile = File("input/2023/day4/input")
-    partOne(inputFile)
-//    partTwo(inputFile)
+//    partOne(inputFile)
+    partTwo(inputFile)
+}
+
+fun partTwo(inputFile: File) {
+    val lineRegex = ":\\s+".toRegex()
+    val cardRegex = "\\s+\\|\\s+".toRegex()
+    val numbersRegex = "\\s+".toRegex()
+
+    val cardsCountMap = mutableMapOf<Int, Int>()
+
+    inputFile.forEachLine { line ->
+        val cardAndNumbers = line.split(lineRegex)
+
+        val cardNumber = cardAndNumbers[0].split(numbersRegex)[1].toInt()
+        val cardsCount = cardsCountMap.merge(cardNumber, 1) { oldValue, _ -> oldValue + 1 }!!
```

Comment:
**Suggestion:** You are using `!!` operator in `val cardsCount = cardsCountMap.merge(cardNumber, 1) { oldValue, _ -> oldValue + 1 }!!`. This can lead to a NullPointerException if the merge operation returns null. Consider using the Elvis operator `?:` to provide a default value when the merge operation returns null. [possible issue]
```suggestion
        val cardsCount = cardsCountMap.merge(cardNumber, 1) { oldValue, _ -> oldValue + 1 } ?: 0
```